batch-nft-reveal/app/components/create/Create.tsx

```diff
@@ -0,0 +1,272 @@
+import React from 'react'
+import { useRouter } from 'next/router'
+import { Field, FormikProvider, useFormik } from 'formik'
+import {
+  Button,
+  Container,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  Tooltip,
+  Divider,
+  Center,
+} from '@chakra-ui/react'
+import { useEthers } from '@usedapp/core'
+// TODO: uncomment when connect with deploy function
+// import { ethers } from 'ethers'
+// import { deployNFTCollection } from '../../lib/deploy'
+// import NFTCollectionParams from '../../lib/types/NFTCollectionParams'
+
+export const Create = (): JSX.Element => {
+  const { account, chainId, library, error } = useEthers()
+  const router = useRouter()
+
+  const formik = useFormik({
+    initialValues: {
+      tokenName: '',
+      symbol: '',
+      maxSupply: '',
+      mintCost: '',
+      revealBatchSize: '',
+      revealInterval: '',
+      vrfSubscriptionId: '',
+    },
+    onSubmit: (values) => {
+      console.log(`Send create request with params: ${values.tokenName}, ${values.symbol}, ${values.maxSupply}, 
+        ${values.mintCost}, ${values.revealBatchSize}, ${values.revealInterval}, ${values.vrfSubscriptionId}`)
+      // TODO: call function for deploy contract here
+      // const nftCollectionParams: NFTCollectionParams = {
+      //   nftName: values.tokenName,
+      //   nftSymbol: values.symbol,
+      //   nftMaxSupply: values.maxSupply,
+      //   nftMintCost: ethers.utils.parseEther(values.mintCost),
+      //   nftRevealBatchSize: values.revealBatchSize,
+      //   nftRevealInterval: values.revealInterval,
+      //   vrfSubscriptionId: values.vrfSubscriptionId,
+      // }
+
+      // const address = deployNFTCollection( // const contract = ...
+      //   nftCollectionParams,
+      //   library?.getSigner(),
+      //   chainId
+      // )
+
+      // router.push(`/collection/${address}`); // /collection/${contract.address}
+    },
+  })
+
+  const isNotEmpty = (value: string): string => {
+    let error: string
+    if (!value) {
+      error = 'is required'
+    }
+    return error
+  }
+
+  const isNotNumber = (value: string) => {
+    let error: string
+    if (!value) {
+      error = 'is required'
+    } else if (!value || !/^[0-9]*$/.test(value)) {
+      error = 'must be a positive number'
+    }
+    return error
+  }
+
+  const isNotPositiveNumber = (value: string) => {
+    let error: string = isNotNumber(value)
+    if (!error && Number(value) <= 0) {
+      error = 'must be greater then zero'
+    }
+    return error
+  }
+
+  const isNotFractualNumber = (value: string) => {
+    let error: string
+    if (!value) {
+      error = 'is required'
+    } else if (!value || !/^(0|([1-9][0-9]*))(\.[0-9]{0,18})?$/.test(value)) {
+      error = 'must be positive fractional number'
+    }
+    return error
+  }
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <Container maxW="mt">
+        <FormikProvider value={formik}>
```

Comment:
What's the purpose of this provider?

Reply:
Without `FormikProvider` the validations in formik will not be working.
`const formik = useFormik(...)` creates the state for formik with all it's functionality.
The provider connects the variable `formik` to the web code inside it.

There are other ways to use formik without `FormikProvider` but with the provider you can use other states and hooks which are not working with some of the other ways you can use formik.